src/Catty/Parser&Serialializer/ProjectMigrator.swift

```diff
@@ -0,0 +1,111 @@
+/**
+ *  Copyright (C) 2010-2020 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+class ProjectMigrator: ProjectMigratorProtocol {
+
+    private let correctVersionNumber: Float = 0.992
+    private var projectPath: String = ""
+    private var directoryPath: String = ""
+
+    enum FileSystem: String {
+        case automaticScreenShot = "automatic_screenshot.png"
+        case maualScreenShot = "manual_screenshot.png"
+        case screenShot = "screenshot.png"
+        case imagesDirectory = "images"
+        case soundDirectory = "sounds"
+    }
+
+    func migrateToScene(project: Project) throws {
+
+        guard let versionNumber = Float(project.header.catrobatLanguageVersion) else {
+            throw ProjectMigratorError.unknown(description: "Unable to convert version number to Float")
+        }
+
+        if versionNumber < self.correctVersionNumber {
+            throw ProjectMigratorError.invalidCatrobatLanguageVersion
+        }
+
+        self.projectPath = project.projectPath()
+        let fileManager = CBFileManager()
```

Comment:
Do not use this dependency like that directly in the middle of your production code. This makes this function untestable. Use constructor dependency injection and assign this dependency to a member variable. That enables you to replace the production version with a test double while unit testing this code.